app/models/govuk_publishing_components/component_doc.rb

```diff
@@ -0,0 +1,34 @@
+module GovukPublishingComponents
+  ComponentDoc = Struct.new(:id, :name, :description, :body, :fixtures) do
```

Comment:
This class has a lot of collection-level functionality, which feels like it should belong in another separate singleton class that has the ability to memoize things properly, and deliver things from hashes rather than performing `find` operations.